Python/Product/PythonTools/PythonTools/Intellisense/TaskProvider.cs

```diff
@@ -461,79 +449,76 @@ abstract class TaskProvider : IVsTaskProvider, IDisposable {
-            var cts = new CancellationTokenSource();
             bool changed = false;
             var lastUpdateTime = DateTime.Now;
 
-            try {
-                // First time through, we don't want to abort the queue. There
-                // should be at least one message or the worker would not have
-                // been started.
-
-                var wh = new[] { cts.Token.WaitHandle, _workerQueueChanged.WaitHandle};
-
-                while (WaitHandle.WaitAny(wh) == 0) {
-                    // Prevent timeouts while processing the message
-                    cts.CancelAfter(-1);
+            while (_workerQueueChanged.Wait(1000)) {
+                // Reset event and handle all messages in the queue
+                _workerQueueChanged.Reset();
 
-                    // Reset event and handle all messages in the queue
-                    _workerQueueChanged.Reset();
-
-                    while (!cts.IsCancellationRequested) {
-                        WorkerMessage msg;
-                        lock (_workerQueue) {
-                            if (_workerQueue.Count == 0) {
-                                break;
-                            }
-                            msg = _workerQueue.Dequeue();
-                        }
-
-                        if (msg is WorkerMessage.AbortMessage) {
-                            // Ignore flush messages and cancel the loop
-                            flushMessages.Clear();
-                            cts.Cancel();
+                while (true) {
+                    WorkerMessage msg;
+                    lock (_workerQueue) {
+                        if (_workerQueue.Count == 0) {
                             break;
-                        } else if (msg is WorkerMessage.FlushMessage) {
-                            // Keep flush messages until we've exited the loop
-                            flushMessages.Enqueue(msg);
-                        } else {
-                            // Apply the message to our collection
-                            changed |= msg.Apply(_items, _itemsLock);
-                        }
-
-                        // Every second, we want to force another update
-                        if (changed) {
-                            var currentTime = DateTime.Now;
-                            if ((currentTime - lastUpdateTime).TotalMilliseconds > 1000) {
-                                Refresh();
-                                lastUpdateTime = currentTime;
-                                changed = false;
-                            }
                         }
+                        msg = _workerQueue.Dequeue();
                     }
 
-                    if (cts.IsCancellationRequested) {
-                        break;
+                    if (msg is WorkerMessage.FlushMessage) {
+                        // Keep flush messages until we've exited the loop
+                        flushMessages.Enqueue(msg);

```

Comment:
I don't see flushMessages getting cleaned up.  Is it ok to just let that build infinitely?